File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/IntegerAttribute.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents an attribute with an integer value.
 */
public class IntegerAttribute extends Attribute<Long> {
  /**
   * Creates a new IntegerAttribute instance with the provided key and value.
   * @param id
   * @param value
   */
  public IntegerAttribute(String id, Long value) {
    super(id, value);
  }

  public static IntegerAttribute fromProto(IdentityEntitiesV2.IntegerAttributeV2 res) {
    return new IntegerAttribute(res.getId(), res.getValue());
  }

  public IdentityEntitiesV2.IntegerAttributeV2 toProto() {
    return IdentityEntitiesV2.IntegerAttributeV2.newBuilder()
        .setId(this.id)
        .setValue(this.value)
        .build();
  }
}

File: languages/php/Bloock/Entity/IdentityV2/IntegerAttribute.php
<?php

namespace Bloock\Entity\IdentityV2;

class IntegerAttribute extends Attribute
{
    public static function fromProto(\Bloock\IntegerAttributeV2 $res): IntegerAttribute
    {
        return new IntegerAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\IntegerAttributeV2
    {
        $p = new \Bloock\IntegerAttributeV2();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}